packages/bootstrap-vue-next/src/components/BTable/BTable.vue

```diff
@@ -295,7 +295,21 @@ const selectedItemsSetUtilities = {
   For some reason, the reference of the object gets lost. However, when you use an actual ref(), it works just fine
   Getting the reference properly will fix all outstanding issues
   */
-  has: (item: Readonly<TableItem<T>>) => selectedItemsToSet.value.has(item),
+  has: (item: Readonly<TableItem<T>>) => {
+    if (props.primaryKey) {
+      let isSelected = false
+      const pkey = props.primaryKey as keyof TableItem<T>
+      for (const selected of selectedItemsToSet.value) {
+        if (selected[pkey] === item[pkey]) {
+          isSelected = true
+          break
+        }
+      }
+      return isSelected
+    }
+
+    return selectedItemsToSet.value.has(item)
+  },
```

Comment:
```suggestion
  has: (item: Readonly<TableItem<T>>) => {
    if (props.primaryKey) {
      const pkey = props.primaryKey as keyof TableItem<T>
      for (const selected of selectedItemsToSet.value) {
        if (selected[pkey] === item[pkey]) {
          return true
        }
      }
      return false
    }

    return selectedItemsToSet.value.has(item)
  },
```

I didn't think about it before, but this could be more concise.